### PR TITLE
fix: log out button displays loading indicator when clicked

### PR DESCRIPTION
**Pull request by aleespryker**
closes: [HRZ-3197](https://spryker.atlassian.net/browse/HRZ-3197)

---

### Checklist before requesting a code review

- [x] PR title is aligned with [PR titles and conventional commits](https://spryker.atlassian.net/wiki/spaces/FA/pages/3821175291/PR+titles+and+conventional+commits)
- [x] I have performed a self-review of my code and have fixed all jobs failures

[HRZ-3197]: https://spryker.atlassian.net/browse/HRZ-3197?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/picking/user-profile/user-profile.component.ts [hunk 1]
```diff
@@ -107,6 +110,7 @@ export class UserProfileComponent extends LitElement {
   }
 
   protected onLogOut(): void {
+    this.logoutLoading = true;
     this.authService.logout();
```
Comment (supproduction) [inline libs/domain/picking/user-profile/user-profile.component.ts]: Is it possible to get error while logout?
Should we change `logoutLoading` to false then?

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/picking/user-profile/user-profile.component.spec.ts [hunk 2]
```diff
@@ -148,6 +148,12 @@ describe('UserProfileComponent', () => {
         element.renderRoot.querySelector('button')?.click();
       });
 
+      it('should show loading indicator', () => {
+        expect(
+          element.renderRoot.querySelector('oryx-button[loading]')
+        ).not.toBeNull();
```
Comment (Sinucid) [inline libs/domain/picking/user-profile/user-profile.component.spec.ts]:
```suggestion
        expect(element).toContainElement('oryx-button[loading]');
```

File: libs/domain/picking/user-profile/user-profile.component.ts [hunk 3]
```diff
@@ -31,6 +31,9 @@ export class UserProfileComponent extends I18nMixin(LitElement) {
   @state()
   protected loading: boolean | null = null;
 
+  @state()
+  protected logoutLoading: boolean | null = null;
```
Comment (Sinucid) [inline libs/domain/picking/user-profile/user-profile.component.ts]:
```suggestion
  protected logoutLoading = false;
```

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/domain/picking/user-profile/user-profile.component.spec.ts [hunk 4]
```diff
@@ -148,6 +148,12 @@ describe('UserProfileComponent', () => {
...
+      });
+
```
Comment (Sinucid) [inline libs/domain/picking/user-profile/user-profile.component.spec.ts:154]: and also would be nice to test error handling

**Review by Sinucid — CHANGES_REQUESTED** [review]

**Review by Sinucid — APPROVED** [review]

**Review by supproduction — APPROVED** [review]

**Pull request merged**